fqn_decorators/asynchronous.py

```diff
@@ -9,19 +9,27 @@
 
 
 class AsyncDecorator(Decorator):
-    async def __call__(self, *args, **kwargs):
-        self.fqn = self.get_fqn()
-        self.args = args
-        self.kwargs = kwargs
+    def __new__(cls, func=None, **params):
+        def _wrap_func(self, f, *args, **kwargs):
+            return self.__class__(f, **self.params)
```

Comment:
I'm trying to wrap (no pun intended) my head around this: is it still possible to pass `*args` in this case?

Reply:
This is a replacement for `Decorator.__call__(self, *args, **kwargs)` method.
In case decorator is parametrised, `func` is not passed into `__init__` call, but it passed into `__call__` as first argument.